### PR TITLE
feat:reports refactored code to calculate counts

### PR DESCRIPTION
**Pull request by Nkdiaz**
Updated code to remove unused code and to have calculated counts for better ease to calculate total counts of validation notices per date in monthly GTFS Reports

### CODE REVIEW

**Comment by machow** [comment]
@Nkdiaz can you paste the before and after screenshots we discussed?

**Comment by Nkdiaz** [comment]
Reports generated now include a count for validation errors and the report tables now display a count with the updated row structure. I am attaching some before and after screenshots. It is worth mentioning that for this date, there is only one  date and that impacts the arrangement of the files/fields existing table. 
validation-errors-before missing date
es.githubusercontent.com/32888103/145257417-93db27e4-3b84-48eb-b4fc-08973bef7673.png)

**Comment by machow** [comment]
Thanks! I'm noticing there's an issue with `views.gtfs_schedule_fact_daily_feed_files` causing it not to show a column for July 6th. It looks like we need to also partition on `T1.name` on this line: https://github.com/cal-itp/data-infra/blob/main/airflow/dags/gtfs_views/gtfs_schedule_fact_daily_feed_files.sql#L43

Let me sanity check quickly to make sure that's the issue and open a PR. Then, when we regenerate reports it should resolve.

**Review by evansiroky — CHANGES_REQUESTED** [review]
This is all pretty awesome and outputting almost exactly what is expected (aside from that decimal display of the validation count). As noted in a comment, raising a `ValueError` is crashing on a particular feed. 

I have made another PR with some code that at least gets things working that I propose to merge into this branch that gets this working. See https://github.com/cal-itp/reports/pull/59.

**Comment by machow** [comment]
@evansiroky here is a PR that should fix the file counts missing on certain days (those where our extractor failed). https://github.com/cal-itp/data-infra/pull/842

File: reports/report.ipynb [hunk 1]
```diff
@@ -112,10 +119,20 @@
...
+    "    \n",
+    "    If there are no rows, then all dictionary values are None.\n",
+    "    \"\"\"\n",
+    "    \n",
+    "    result = collect(df)\n",
+    "    \n",
+    "    if len(result) == 0:\n",
+    "        return {k: None for k in result.columns}\n",
+    "    elif len(result) > 1:\n",
+    "        raise ValueError(\"table contained more than 1 row\")\n",
```
[the remark this answers is not in the document]
Comment (machow) [reply reports/report.ipynb]: ah, thanks--I think it uses the agency.txt table to connect back to a specific feed_info file as a means of distinguishing the feeds it aggregates :/

**Review by machow — APPROVED** [review]

**Comment by machow** [comment]
Punted my comments about file checks into a issue, so if useful, we can merge these in and work any useful next steps there: https://github.com/cal-itp/reports/issues/62

**Comment by evansiroky** [comment]
Thanks for isolating #62. As noted in that issue, let's just hide the columns for now.

However, it would still be good to get the validation errors using the fact tables as mentioned in https://github.com/cal-itp/reports/issues/63.

**Review by evansiroky — APPROVED** [review]

**Pull request merged**